### PR TITLE
Refactor highlight and horizontal layout

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-NNNN]

### Description

In terms of making migration of OutlineInput and related components easier, I decided to refactor how we use `highlight` and `layout` props that we pass to OutlineInput. The purpose is to reduce dependencies between form components.

### How to test

- [Temploy](https://picasso.toptal.net/fx-null-highlight)
- No visual changes

### Development checks

- [#] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- Annotate all `props` in component with documentation
- Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- codemod is created and showcased in the changeset
- test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

### CODE REVIEW

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

File: packages/base/NumberInput/src/NumberInput/styles.ts [hunk 1]
```diff
@@ -1,8 +1,7 @@
-import type { Theme } from '@material-ui/core/styles'
+// import type { Theme } from '@material-ui/core/styles'
```
Comment (ruslan-sed) [inline packages/base/NumberInput/src/NumberInput/styles.ts:1]: This probably was left here accidentaly

File: packages/base/OutlinedInput/src/OutlinedInput/styles.ts [hunk 2]
```diff
@@ -35,6 +35,8 @@ PicassoProvider.override(
           color: palette.grey.main,
         },
 
+        // '&$highlightAutofill': {},
```
Comment (ruslan-sed) [inline packages/base/OutlinedInput/src/OutlinedInput/styles.ts:38]: Do we need this?

File: packages/base/Select/src/NonNativeSelect/styles.ts [hunk 3]
```diff
@@ -40,7 +38,7 @@ export default (theme: Theme) => {
       outline: 0,
     },
     outlinedInput: {
-      backgroundColor: palette.common.white,
+      // backgroundColor: palette.common.white,
```
Comment (ruslan-sed) [inline packages/base/Select/src/NonNativeSelect/styles.ts:41]: Do we need this?

File: packages/base/OutlinedInput/src/OutlinedInput/OutlinedInput.tsx [hunk 4]
```diff
@@ -190,8 +197,13 @@ const OutlinedInput = forwardRef<HTMLElement, Props>(function OutlinedInput(
...
           classes[`root${capitalize(width)}`],
           classes[`root${capitalize(size)}`],
-          { [`${classes.hidden}`]: type === 'hidden' },
-          { [classes.rootDark]: isDark }
+          {
+            [`${classes.hidden}`]: type === 'hidden',
+            [classes.rootDark]: isDark,
+            [classes.highlightAutofill]: highlight === 'autofill',
+            [classes.horizontalLayout]: layout === 'horizontal',
+            [classes.error]: Boolean(status === 'error' || error),
```
Comment (ruslan-sed) [inline packages/base/OutlinedInput/src/OutlinedInput/OutlinedInput.tsx:205]: This was not moved here from some of parent components, as `horizontalLayout` or `highlightAutofill`. Where this came from?

**Review by ruslan-sed — APPROVED** [review]
Good idea to move those styles handling into `OutlinedInput` 👍
Left a few small comments.

File: packages/base/OutlinedInput/src/OutlinedInput/OutlinedInput.tsx (hunk 4 above)
Comment (TomasSlama) [reply packages/base/OutlinedInput/src/OutlinedInput/OutlinedInput.tsx:205]: Good question!
So this `error` is related to [white background on select](https://github.com/toptal/picasso/pull/4317/files#diff-ddb97468335305c7f8d6545435642c5763508adfa6e0d5610cf2742e8a3eb2d8L43). I was checking why it is needed on Select level and I found an old PR that was fixing a bug, where the Select had wrong background color when in error state. It was visible only on a different background than white. 

So I moved the solution to the OutlinedInput and solved it also for all other inputs.

File: packages/base/Select/src/NonNativeSelect/styles.ts (hunk 3 above)
Comment (TomasSlama) [reply packages/base/Select/src/NonNativeSelect/styles.ts]: thanks, removed all leftovers 🙇

**Comment by TomasSlama** [comment]
@toptal-anvil ping reviewers

**Pull request merged**